Interfaces/API/Job.py

```diff
@@ -1107,39 +1105,51 @@ def _toJDL( self, xmlFile = '', jobDescriptionObject = None ):  # messy but need
         paramsDict['InputData']['value'] = extraFiles
         paramsDict['InputData']['type'] = 'JDL'
 
-    # Handle here the Parametric values
-    if self.parametric:
-      for pType in ['InputData', 'InputSandbox']:
-        if self.parametric.has_key( pType ):
-          if paramsDict.has_key( pType ) and paramsDict[pType]['value']:
-            pData = self.parametric[pType]
-            # List of lists case
-            currentFiles = paramsDict[pType]['value'].split( ';' )
+    # Handle parameter sequences
+    if self.numberOfParameters > 0:
```

Comment:
If it was me, just to be rock solid, I would put what follows this "if" in a separate function/method, and write a comprehensive unit test for it: the number of `if` and `isinstance`, and `types` and `values` inside make it otherwise very difficult to be tested properly by "just submitting jobs" with it.